common/src/main/java/io/netty/util/internal/MacAddressUtil.java

```diff
@@ -227,20 +227,23 @@ private static int compareAddresses(byte[] current, byte[] candidate) {
         }
 
         // Prefer globally unique address.
-        if (current.length == 0 || (current[0] & 2) == 0) {
-            if ((candidate[0] & 2) == 0) {
+        if ((candidate[0] & 2) == 0) {
+            if (current.length != 0 && (current[0] & 2) == 0) {
                 // Both current and candidate are globally unique addresses.
                 return 0;
             } else {
                 // Only current is globally unique.
```

Comment:
Yes, those got swapped when I reversed the conditions. I pushed 0993c6282b55bcd7aff3fed6b3b79cd0dea59c3e which added more tests too. Thanks.